hazelcast/src/main/java/com/hazelcast/map/impl/operation/PutFromLoadAllOperation.java

```diff
@@ -70,8 +71,12 @@ public void run() throws Exception {
             Object previousValue = recordStore.putFromLoad(key, value);
 
             callAfterPutInterceptors(value);
+            // Null-pointer here, in case of null key loaded by MapLoader
+            checkNotNull(key, "Key loaded by a MapLoader cannot be null.");
```

Comment:
Shouldn't we make this key check & value check before everything else, even before `recordStore.putLoad`, which internally checks null keys & values. Then we can remove the extra check inside `DefaultRecordStore.putFromLoadInternal`. Currently is key/value is null, we still call afterPut Interceptor, but this data won't be added to map.